src/api/spec/controllers/webui/kiwi/images_controller_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'rails_helper'
+
+# WARNING: Some tests require real backend answers, so make sure you uncomment
+# this line and start a test backend.
+# CONFIG['global_write_through'] = true
+
+RSpec.describe Webui::Kiwi::ImagesController, type: :controller, vcr: true do
+  let(:project) { create(:project, name: 'fake_project') }
+
+  describe 'GET #import_from_package' do
+    context 'without a kiwi file' do
+      let(:package) { create(:package, name: 'package_without_kiwi_file', project: project) }
+
+      before do
+        get :import_from_package, params: { package_id: package.id }
+      end
+
+      it { expect(response).to redirect_to(root_path) }
+      it { expect(flash[:error]).to eq('There is no KIWI file') }
+    end
+
+    context 'with a kiwi file' do
+      context 'with a valid kiwi file' do
+        let(:kiwi_image_with_package_with_kiwi_file) do
```

Comment:
[Nitpick] you can shorten the variable name to just `kiwi_image` and then expand the context to:
`context 'with a valid kiwi file with a package`